### DISCUSSION TITLE
Adding a new layer to an image

### DISCUSSION THREAD

**Discussion by avialonn** [ImageMagick/ImageMagick#2679] [Help]
Hi there,
How can I add an image on to another image?
I need to manipulate both images differently. 
The first must be grayscale and the second (top) must be pink with opacity.
Anything I tried was producing the same results where the manipulation of the first image always override the second hence, the end image was a grayscale image.
I've tried both Composite and MagickImageCollection methods.
Will appreciate any input on this.
Thanks
aa

`

**Comment by fmw42** [comment]
What is your IM version and platform? What was your exact command using composite? Can you post your input images?

**Comment by avialonn** [comment]
@fmw42 moved this to the appropriate forum. thanks!

**Reply by fmw42** [reply]
Please answer my questions and provide your images so we can try to help you

**Reply by avialonn** [reply]
I actually did but just realized that this was not the appropriate place to post my question. Im using the .net core wrapper hence moved it to there. thanks again!

**Comment by snibgo** [comment]
Please provide your input images 1.jpeg and pink.png. Otherwise, we can only guess.

> I've tried this code with no success:...

What does that mean? Does your program crash, or what?

Your two samples of code do entirely different things. The first composites pink.png on 1.jpeg, and the second does this in the opposite order.
